serialfile.go

```diff
@@ -47,7 +58,7 @@ func NewSerialFile(path string, hidden bool, stat os.FileInfo) (Node, error) {
 		if err != nil {
 			return nil, err
 		}
-		return &serialFile{path, contents, stat, hidden}, nil
+		return &serialFile{path, filterFiles(filter, contents), stat, filter}, nil
```

Comment:
Don't we end up filtering twice here? We filter out hidden files again when we walk the directory tree.

I'd just keep the lazy filtering and replace lines like:

```go
for !it.filter.IncludeHidden && isHidden(stat) {
```

With:

```go
for !it.filter.ShouldExclude(stat) {
```

Reply:
If contents is filtered at construction, then filtering only happens once, because the serialFile object can only iterate over fileInfo slice passed into it. I went with filtering at when iterating over a serialFile's entries.